torch/utils/_sympy/value_ranges.py

```diff
@@ -830,12 +953,13 @@ def cast(x, dtype):
 
     @staticmethod
     def square(x):
-        return ValueRanges.convex_min_zero_map(x, lambda y: y * y)
+        return ValueRanges.convex_min_zero_map(x, lambda y: PowByNatural(y, 2))
 
     @staticmethod
     def neg(x):
         return ValueRanges.decreasing_map(x, operator.neg)
 
+    # TODO: this is wrong, do something better
```

Comment:
can you elaborate?

Reply:
```
    # TODO: this is slightly inaccurate because truncdiv operates at integer
    # precision, but we're going through float truediv which means we can     
    # potentially lose precision on the bounds
```